File: week4/p5.c
#include <stdio.h>

int josephus(int n, int k) {
    int res = 0;
    for (int i = 1; i <= n; i++) {
        res = (res + k) % i;
    }
    return res + 1;
}

int main() {
    int n = 5, k = 3;
    printf("%d\n", josephus(n, k));
    return 0;
}

File: week4/p5.py
def josephus(n, k):
    res = 0
    for i in range(1, n + 1):
        res = (res + k) % i
    return res + 1

if __name__ == "__main__":
    n, k = 5, 3
    print(josephus(n, k))
